app/src/androidTest/java/com/duckduckgo/app/browser/BrowserTabViewModelTest.kt

```diff
@@ -2460,7 +2462,6 @@ class BrowserTabViewModelTest {
         givenDeviceLocationSharingIsEnabled(true)
         givenLocationPermissionIsEnabled(true)
         givenCurrentSite(domain)
-
         testee.onSiteLocationPermissionSelected(domain, LocationPermissionType.ALLOW_ONCE)
```

Comment:
Hey @malmstein, This test `whenDomainRequestsSitePermissionAndUserAllowedSessionPermissionThenPermissionIsAllowed` is failing for me.

I took a look but I'm not familiar with the logic here. What I've seen is:
- when calling `onSiteLocationPermissionSelected` I guess we expect to simulate that for that domain a permission is selected. However the method does nothing because `locationPermission?.let {` -> locationPermission is `null` at that point.
- When we call `givenLocationPermissionIsEnabled`, `CheckSystemLocationPermission` is emitted then.

could you take a look? Thanks!

Reply:
Discussed online with @cmonfortep and refactored the test to verify the functionality.